plugin/advisor/catalog/walk_through.go

```diff
@@ -0,0 +1,427 @@
+package catalog
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/bytebase/bytebase/plugin/advisor/db"
+	"github.com/bytebase/bytebase/plugin/parser"
+
+	tidbparser "github.com/pingcap/tidb/parser"
+	tidbast "github.com/pingcap/tidb/parser/ast"
+	"github.com/pingcap/tidb/parser/format"
+	"github.com/pingcap/tidb/parser/model"
+)
+
+// WalkThroughErrorType is the type of WalkThroughError.
+type WalkThroughErrorType int
+
+const (
+	// PrimaryKeyName is the string for PK.
+	PrimaryKeyName string = "PRIMARY"
+	// FullTextName is the string for FULLTEXT.
+	FullTextName string = "FULLTEXT"
+
+	// ErrorTypeEngineTypeError is the error for unsupported engine type.
+	ErrorTypeEngineTypeError WalkThroughErrorType = 1
```

Comment:
I would generalize this as Unsupported error ode, especially since we are using the premium single-digit error code.